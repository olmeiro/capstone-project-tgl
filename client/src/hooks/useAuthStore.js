import React, { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { clearErrorMessage, onChecking, onLogin, onLogout, onLoadDataProfile } from '../store'

export const useAuthStore = () => {
  const { status, user, errorMessage } = useSelector(state => state.auth)

  const dispatch = useDispatch()

  const startLogin = async ({ alias, password }) => {
    dispatch(onChecking())
    try {
      const { data } = await socialApi.post('/user/login', { alias, password })
      localStorage.setItem('token', data.body.token)

      const id = data.body.user.id
      const aliasUser = data.body.user.alias
      const name = data.body.user.name
      const bio = data.body.user.bio
      const email = data.body.user.email
      const phone = data.body.user.phone
      const photoProfile = data.body.user.photoProfile
      const favorites = data.body.user.favorites
      const friends = data.body.user.friends

      dispatch(onLoadDataProfile({ id, aliasUser, name, bio, email, phone, photoProfile, favorites, friends }))
      dispatch(onLogin({ id: data.body.user.id, alias: data.body.user.alias, name: data.body.user.name, photoProfile: data.body.user.photoProfile, friends: data.body.user.friends }))
    } catch (error) {
      dispatch(onLogout(error.response.data?.message || ''))
      setTimeout(() => {
        dispatch(clearErrorMessage())
      }, 10)
    }
  }

  const startRegister = async ({ alias, name, email, phone, password }) => {
    dispatch(onChecking())
    try {
      const { data } = await socialApi.post('/user', { alias, name, email, phone, password })
      localStorage.setItem('token', data.body.token)
      dispatch(onLogin({ id: data.body.user.id, alias: data.body.user.alias, name: data.body.user.name }))
    } catch (error) {
      dispatch(onLogout(error.response.data?.message || ''))
      setTimeout(() => {
        dispatch(clearErrorMessage())
      }, 10)
    }
  }

  const checkToken = async () => {
    const token = localStorage.getItem('token')
    if (!token) {
      return dispatch(onLogout())
    }

    try {
      const { data } = await socialApi('/user/renew')
      localStorage.setItem('token', data.body.token)

      const dataUser = await socialApi.get(`/user/byid/${data.body.id}`)

      const id = dataUser.data.body.id
      const aliasUser = dataUser.data.body.alias
      const name = dataUser.data.body.name
      const bio = dataUser.data.body.bio
      const email = dataUser.data.body.email
      const phone = dataUser.data.body.phone
      const photoProfile = dataUser.data.body.photoProfile
      const favorites = dataUser.data.body.favorites
      const friends = dataUser.data.body.friends

      dispatch(onLoadDataProfile({ id, aliasUser, name, bio, email, phone, photoProfile, favorites, friends }))
      dispatch(onLogin({ id: data.body.id, alias: data.body.alias, name: data.body.name }))
    } catch (error) {
      localStorage.clear()
      dispatch(onLogout())
    }
  }

  const startLogout = () => {
    localStorage.clear()
    dispatch(onLogout())
  }

  return {
    // properties
    errorMessage,
    status,
    user,
    // methods
    startLogin,
    startRegister,
    checkToken,
    startLogout
  }
}
