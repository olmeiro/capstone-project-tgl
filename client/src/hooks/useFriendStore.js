import { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { getFriendsFromUser, deleteFriend } from '../store'

export const useFriendStore = () => {
  const dispatch = useDispatch()
  const { user } = useSelector(state => state.auth)
  const userId = user.id

  const getFriendsFromUserHook = async () => {
    const response = await socialApi.get(`/friends/${userId}`)
    const friends = response.data.body
    dispatch(getFriendsFromUser(friends))
  }
  const deleteFriendHook = async friendId => {
    await socialApi.delete(`/friends?userid=${userId}&friendid=${friendId}`)
    dispatch(deleteFriend(friendId))
  }
  return {
    getFriendsFromUserHook,
    deleteFriendHook
  }
}
