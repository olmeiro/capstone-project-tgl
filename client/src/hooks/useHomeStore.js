import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { checkFavorites, getFavorites, actionKeepLastUserVisitesOnRedux, getPostsToHome, getFriendsFromFriends, checkComments, getUserSearched, checkEmptySearchBar, setPathReference, setLastUserVisited, setInfoFromUserLoggedIn, setcheckInfoUser, setPostsFromUserLoggedIn } from '../store'

export const useHomeStore = () => {
  const dispatch = useDispatch()
  const { user } = useSelector(state => state.auth)
  const userId = user.id

  const entire = socialApi.get(`/user/byid/${userId}`).then(response => response.data.body)
  let friends = entire.then(user => user.friends)

  const getPostsToHomeHook = async () => {
    friends = await friends
    let postsTotalFromFriends = friends.map(async friendId => await socialApi.get(`/posts/byuser/${friendId}`))
    postsTotalFromFriends = Promise.all(postsTotalFromFriends)
    postsTotalFromFriends = await postsTotalFromFriends
    postsTotalFromFriends = postsTotalFromFriends.map(posts => posts.data.body).flat()
    dispatch(getPostsToHome(postsTotalFromFriends))
  }
  const getFriendsFromFriendsHook = async () => {
    const wholeUser = await socialApi.get(`/user/byid/${userId}`)
    const friendsId = wholeUser.data.body.friends
    let suggestion = friendsId.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body.friends
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    suggestion = suggestion.flat()
    suggestion = suggestion.filter((friendid, index) => suggestion.indexOf(friendid) == index)
    suggestion = suggestion.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    dispatch(getFriendsFromFriends(suggestion))
  }
  const addFriendshipHook = async (friendId) => {
    await socialApi.post('/friends/', { userId, friendId })
  }
  const likeAPost = async (id) => {
    await socialApi.put('/posts/', { id, likes: 3 })
  }
  const makeAComment = async (postId, comment) => {
    await socialApi.post('/comments/', { comment, postId, userId })
  }
  const getCommentsByPost = async (postid) => {
    let comments = await socialApi.get(`/comments/bypost/${postid}`)
    comments = comments.data.body
    comments = comments.map(async comment => {
      const userId = comment.UserId
      let user = await socialApi.get(`/user/byid/${userId}`)
      user = user.data.body
      comment.user = user
      return comment
    })
    comments = await Promise.all(comments)
    return comments
  }
  const deleteComment = async (commentId) => {
    await socialApi.delete(`/comments/${commentId}`)
  }
  const checkCommentsHook = () => {
    dispatch(checkComments())
  }
  const searchUserByAlias = async (aliasFromBody) => {
    let foundUsers = null
    if (aliasFromBody.trim() != '') {
      const response = await socialApi.get('/user/all')
      const users = response.data.body
      foundUsers = users.filter(user => user.alias.toLowerCase().includes(aliasFromBody && aliasFromBody.toLowerCase()))
    }
    dispatch(getUserSearched(foundUsers))
  }
  const checkEmptySearchBarHook = () => {
    dispatch(checkEmptySearchBar())
  }
  const sendPathHook = (pathReference) => {
    dispatch(setPathReference(pathReference))
  }
  const setLastUserVisitedHook = async (userAlias, userId) => {
    const responseUser = await socialApi.get(`user/byalias/${userAlias}`)
    const user = responseUser.data.body
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    user.posts = posts
    localStorage.setItem('lastUserVisited', JSON.stringify(user))
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(setLastUserVisited(lastUserVisited))
  }
  const keepLastUserVisitesOnRedux = async () => {
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(actionKeepLastUserVisitesOnRedux(lastUserVisited))
  }
  const getInfoFromTheUserLoggedIn = async () => {
    const user = await entire
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    const newUser = {
      ...user,
      posts
    }
    dispatch(setInfoFromUserLoggedIn(newUser))
  }
  const checkInfoUserHook = async () => {
    dispatch(setcheckInfoUser())
  }
  const getPostsFromUserLoggedIn = async () => {
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    dispatch(setPostsFromUserLoggedIn(posts))
  }
  const addToFavorites = async (postFavoriteId, userLogeadoId) => {
    await socialApi.post(`/favorites?postId=${postFavoriteId}&userId=${userLogeadoId}`)
  }
  const getFavoritesHook = async () => {
    const response = await socialApi.get(`/favorites/${userId}`)
    const favorites = response.data.body
    dispatch(getFavorites(favorites))
  }
  const deleteFavoriteHook = async (postId) => {
    await socialApi.delete(`/favorites?postId=${postId}&userId=${userId}`)
  }
  const checkFavoritesHook = async () => {
    dispatch(checkFavorites())
  }

  return {
    getPostsToHomeHook,
    getFriendsFromFriendsHook,
    addFriendshipHook,
    likeAPost,
    makeAComment,
    getCommentsByPost,
    deleteComment,
    checkCommentsHook,
    searchUserByAlias,
    checkEmptySearchBarHook,
    sendPathHook,
    setLastUserVisitedHook,
    getInfoFromTheUserLoggedIn,
    checkInfoUserHook,
    getPostsFromUserLoggedIn,
    keepLastUserVisitesOnRedux,
    addToFavorites,
    getFavoritesHook,
    deleteFavoriteHook,
    checkFavoritesHook
  }
}
