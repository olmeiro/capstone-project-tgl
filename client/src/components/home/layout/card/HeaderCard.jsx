import React, { useEffect, useState } from 'react'
import {
  StarIcon,
  HeartIcon,
  PlusCircleIcon,
  ThumbUpIcon
} from '@heroicons/react/outline'
import { Tooltip, Button, Modal, Label, Textarea } from 'flowbite-react'
import { useHomeStore } from '../../../../hooks/useHomeStore'
import Swal from 'sweetalert2'
import { socialApi } from '../../../../api'
import { useSelector } from 'react-redux'

export default function HeaderCard({ setIsOpen, photo, description, likes, date, likeAPost, postId, userId }) {
  const [openModal, setOpenModal] = useState(false)
  const [likesRender, setLikesRender] = useState(likes)
  const [comment, setComment] = useState('')
  const { makeAComment, checkCommentsHook, addToFavorites } = useHomeStore()
  const [userOfPost, setUserOfPost] = useState()
  const { user } = useSelector(state => state.auth)

  let entireUser = socialApi.get(`/user/byid/${userId}`).then(response => response.data.body)
  useEffect(() => {
    (async () => {
      entireUser = await entireUser
      setUserOfPost(entireUser)
    })()
  }, [])

  const handleAddFavorite = () => {
    addToFavorites(postId, user.id)
    Swal.fire('¡Publicación agregada a favoritos!')
  }

  const handleLike = () => {
    likeAPost(postId)
  }
  const handleComment = (e) => {
    setComment(e.target.value)
  }
  const handleSumit = async () => {
    await makeAComment(postId, comment)
    setComment('')
    checkCommentsHook()
    Swal.fire('Comentario publicado')
    setOpenModal(false)
  }

  return (
    <div>
      <React.Fragment>
        <Modal show={openModal} size="md" popup={true} onClose={() => setOpenModal(false)}>
          <Modal.Header />
          <Modal.Body>
            <div className="space-y-6 px-6 pb-4 sm:pb-6 lg:px-8 xl:pb-8">
              <h3 className="text-xl font-medium text-gray-900 dark:text-white">
                Deja tu comentario
              </h3>
              <div>
                <div className="mb-2 block">
                  <Label
                    htmlFor="comment"
                    value="Comentario"
                  />
                </div>
                <Textarea
                  id="comment"
                  placeholder="Escribe el comentario"
                  required={true}
                  onChange={(e) => handleComment(e)}
                  value={comment}
                />
              </div>
              <div className="w-full">
                <Button onClick={() => handleSumit()}>
                  Enviar
                </Button>
              </div>
            </div>
          </Modal.Body>
        </Modal>
      </React.Fragment>

      <div className="bg-team-blue rounded-lg ">
        <div className='flex items-center rounded'>
            <img className="h-16 w-16 p-1 rounded-full" src={userOfPost && userOfPost.photoProfile} alt="" />
            <div>
              <p className='underline leading-3 text-lg antialiased font-extrabold'>{userOfPost && userOfPost.alias.toUpperCase()}</p>
              <p className='text-xs leading-5 italic mt-2'>{ date }</p>
            </div>
        </div>
        <div className='text-left ml-16 italic text tracking-wide'>
          {description}
        </div>
        <div className=" box-content max-w-md mx-auto rounded-br-lg rounded-bl-lg  shadow-md overflow-hidden md:max-w-2xl mt-2">
          <div className=' flex justify-center'>
            <img
              className="sm:h-48 sm:w-full object-cover md:h-80 md:w-full "
              src={photo}
              alt="Man looking at item at a store"
            />
          </div>
          <header className="flex items-center p-4">
            <button
              className="bg-team-dark w-1/3 text-team-green sm:text-sm sm:mr-2 rounded px-4 py-2 hover:text-white"
              onClick={() => setIsOpen(true)}
            >
              Ver comentarios
            </button>
            <div className="flex flex-row w-2/3 justify-end items-center gap-4">
              <Tooltip content="likes" arrow={false}>
                <div>
                  {likesRender}
                </div>
              </Tooltip>
              <Tooltip content="like" arrow={false}>
                <button onClick={() => handleLike()}>
                  <ThumbUpIcon
                    className="h-6 w-6 mb-3 relative top-1  rounded-md  hover:fill-black"
                    aria-hidden="true"
                  />
                </button>
              </Tooltip>
              <Tooltip content="guardar" arrow={false}>
                <button onClick={() => handleAddFavorite()}>
                  <StarIcon
                    className="h-6 w-6 mb-3 relative top-1  rounded-md hover:fill-black"
                    aria-hidden="true"
                  />
                </button>
              </Tooltip>
              <Tooltip content="comentar" arrow={false}>
                <button>
                  <PlusCircleIcon
                    onClick={() => setOpenModal(true)}
                    className="h-6 w-6 mb-3 relative top-1  rounded-md  hover:fill-black hover:text-white"
                    aria-hidden="true"
                  />
                </button>
              </Tooltip>
            </div>
          </header>
        </div>
      </div>
    </div>
  )
}
