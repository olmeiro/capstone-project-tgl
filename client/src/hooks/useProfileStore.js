import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import Swal from 'sweetalert2'

import { socialApi } from '../api'
import {
  onLoadDataProfile,
  onChangeDataProfile,
  onLoadCommentPhoto,
  inactivatingCount,
  deletingCount,
  onLoadFriendsUser,
  onLoadPublication,
  onSendPublication,
  onLoadPhotoProfile,
  onChanging
} from '../store'
import { useHomeStore } from './useHomeStore'

export const useProfileStore = () => {
  const profile = useSelector((state) => state.profile)
  const dispatch = useDispatch()
  const { getPostsFromUserLoggedIn } = useHomeStore()
  const loadingDataProfile = async (id) => {
    try {
      const { data } = await socialApi.get(`/user/byid/${id}`)
      dispatch(onLoadDataProfile(data.body))
      Swal.fire({
        icon: 'success',
        title: 'Imagen cargada correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo ha ido mal!'
      })
    }
  }

  const loadingFriendsUser = () => {
    // get info api
    dispatch(onLoadFriendsUser()) // pasar info api
  }

  const loadingPhotoProfile = async (image, idUser) => {
    const formData = new FormData()
    formData.append('file', image)
    try {
      await socialApi.put('/user/profilephoto', formData)
      let user = await socialApi.get(`/user/byid/${idUser}`)
      user = user.data.body
      dispatch(onLoadPhotoProfile(user.photoProfile))
      Swal.fire({
        icon: 'success',
        title: 'Fotos de perfil cargada correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo ha ido mal!'
      })
    }
  }

  const sendPublicationUser = async (loginUserId, comment, image) => {
    const formData = new FormData()
    formData.append('loginUserId', loginUserId)
    formData.append('description', comment)
    formData.append('file', image)

    try {
      onLoadChanging()
      const { data } = await socialApi.post('/posts', formData, {
        headers: {
          'Content-Type': 'multipart/form-data'
        }
      })
      getPostsFromUserLoggedIn()
      dispatch(onSendPublication(data.body))
      onLoadChanging()
      Swal.fire({
        icon: 'success',
        title: 'Fotos cargadas correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo a ido mal cargando las imagenes!'
      })
    }
  }

  const loadingPublicationUser = async (id) => {
    try {
      const { data } = await socialApi.get('/posts/all', id)
      dispatch(onLoadPublication(data.body))
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo a ido mal cargando las publicaciones!'
      })
    }
  }

  const changeDataProfile = async ({
    idUser,
    alias,
    name,
    bio,
    email,
    phone
  }) => {
    try {
      await socialApi.put('/user/', { idUser, alias, name, bio, email, phone })
      dispatch(onChangeDataProfile({ alias, name, bio, email, phone }))
      Swal.fire({
        icon: 'success',
        title: 'Información actualizada correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Información no actualizada correctamente.'
      })
    }
  }

  const uploadCommentPhoto = (comment, id) => {
    // TODO: hacer put comment en photo cover de user id
    dispatch(onLoadCommentPhoto(comment))
  }

  const deletePostUser = async (postId, userId) => {
    try {
      Swal.fire({
        title: 'Esta seguro de eliminar publicación?',
        text: 'No puedes revertir esta acción!',
        icon: 'warning',
        showCancelButton: true,
        confirmButtonColor: '#3085d6',
        cancelButtonColor: '#d33',
        confirmButtonText: 'Sí, borrar!'
      }).then(async (result) => {
        if (result.isConfirmed) {
          try {
            await socialApi.delete(`/posts/${postId}`)
            getPostsFromUserLoggedIn()
            loadingPublicationUser(userId)
            Swal.fire(
              'Borrado!',
              'La publicación ha sido eliminada.',
              'success'
            )
          } catch (error) {
            Swal.fire('No se pudo eliminar la publicación.')
          }
        }
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Información no actualizada correctamente.'
      })
    }
  }

  const inactiveCount = (id) => {
    // TODO: change data api user id borrado logico
    dispatch(inactivatingCount(id))
    // onLogout auth
  }

  const deleteCount = (id) => {
    // TODO: change data api user id
    dispatch(deletingCount(id))
  }

  const onLoadChanging = () => {
    dispatch(onChanging())
  }

  return {
    // properties
    profile,
    loadingFriendsUser,
    loadingPhotoProfile,
    sendPublicationUser,
    loadingPublicationUser,
    // methods
    onLoadChanging,
    loadingDataProfile,
    changeDataProfile,
    uploadCommentPhoto,
    deletePostUser,
    inactiveCount,
    deleteCount
  }
}
