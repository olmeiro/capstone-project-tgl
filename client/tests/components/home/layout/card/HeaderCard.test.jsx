import { render, screen, waitFor } from '@testing-library/react'
import HeaderCard from '../../../../../src/components/home/layout/card/HeaderCard'
import { useHomeStore } from '../../../../../src/hooks/useHomeStore'
import { useSelector } from 'react-redux'

jest.mock('../../../../../src/hooks/useHomeStore')

jest.mock('react-redux', () => ({
  useSelector: jest.fn()
}))

describe('test component <HeaderCard />', () => {
  test('should render component', () => {
    useSelector.mockImplementation(() => ('hello world'))
    useHomeStore.mockReturnValue({
      makeAComment: jest.fn(),
      checkCommentsHook: jest.fn(),
      addToFavorites: jest.fn()
    })
    const { container } = render(<HeaderCard />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Ver comentarios')).toBeTruthy()
    // screen.debug()
  })
})
