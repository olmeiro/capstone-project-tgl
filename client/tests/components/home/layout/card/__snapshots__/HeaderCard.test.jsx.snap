// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component <HeaderCard /> should render component 1`] = `
<div>
  <div>
    <div
      class="bg-team-dark rounded-lg "
    >
      <div
        class="flex "
      >
        <div>
          <img
            alt=""
            class="h-16 w-16 rounded-full m-4"
          />
        </div>
        <div>
          <div
            class=" text-lg font-medium text-white mt-6"
          />
          <div
            class=" text-sm text-gray-400 inline-block"
          />
        </div>
      </div>
      <div
        class="ml-6 mb-3 text-sm text-white"
      />
      <div
        class=" box-content max-w-md mx-auto rounded-br-lg rounded-bl-lg bg-myColor shadow-md overflow-hidden md:max-w-2xl mt-2"
      >
        <div
          class=" flex justify-center "
        >
          <img
            alt="Man looking at item at a store"
            class="sm:h-48 sm:w-full object-cover md:h-80 md:w-full "
          />
        </div>
        <header
          class="flex justify-around items-center p-3 bg-team-dark"
        >
          <button
            class="bg-black text-team-green sm:text-sm sm:mr-2 rounded px-4 py-1 hover:bg-black  hover:border-team-green "
          >
            Ver comentarios
          </button>
          <div
            class="flex flex-row w-2/3 justify-end items-center gap-4"
          >
            <div
              class="w-fit"
              data-testid="tooltip-target"
            >
              <button>
                <svg
                  aria-hidden="true"
                  class="h-6 w-6 mb-3 relative top-2  rounded-md hover:fill-team-green  hover:text-black "
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="absolute inline-block rounded-lg py-2 px-3 text-sm font-medium shadow-sm transition-opacity duration-300 invisible opacity-0 bg-gray-900 text-white dark:bg-gray-700"
              data-testid="tooltip"
              id=":r0:"
              role="tooltip"
              style="position: absolute;"
              tabindex="-1"
            >
              <div
                class="relative z-20"
              >
                Agregar a favoritos
              </div>
            </div>
            <div
              class="w-fit"
              data-testid="tooltip-target"
            >
              <button>
                <svg
                  aria-hidden="true"
                  class="h-6 w-6 mb-3 relative top-2  rounded-md  hover:fill-team-green  hover:text-black "
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 9v3m0 0v3m0-3h3m-3 0H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="absolute inline-block rounded-lg py-2 px-3 text-sm font-medium shadow-sm transition-opacity duration-300 invisible opacity-0 bg-gray-900 text-white dark:bg-gray-700"
              data-testid="tooltip"
              id=":r2:"
              role="tooltip"
              style="position: absolute;"
              tabindex="-1"
            >
              <div
                class="relative z-20"
              >
                Comentar
              </div>
            </div>
          </div>
        </header>
      </div>
    </div>
  </div>
</div>
`;
